### PR TITLE
trimming some unnecessary code

### PR DESCRIPTION
**Pull request by lemire**

### CODE REVIEW

File: src/fallback/implementation.cpp [hunk 1]
```diff
@@ -350,15 +350,24 @@ simdutf_warn_unused size_t implementation::maximal_binary_length_from_base64(con
...
 
 simdutf_warn_unused result implementation::base64_to_binary(const char * input, size_t length, char* output) const noexcept {
-  return scalar::base64::base64_to_binary(input, length, output);
+  if(length > 0 && input[length - 1] == '=') {
+    length -= 1;
+    if(length > 0 && input[length - 1] == '=') {
+      length -= 1;
+    }
+  }
+  if(length == 0) {
```
Comment (anonrig) [inline src/fallback/implementation.cpp:359]: If we move this if statement before the previous line, the implementation can be simplified a lot

Comment (lemire) [reply src/fallback/implementation.cpp:359]: Can you elaborate? I am sure that this function can be simplified, but I am not sure what you have in mind. The trick here is that we must discard the padding characters before we can have the 'true' length. Here we decide that if the empty is empty, or contains only one or two padding characters ('='), then we accept it as valid, but return 0 bytes.

Comment (anonrig) [reply src/fallback/implementation.cpp:359]: Actually yes you're right. Resolving this.

**Review by anonrig — APPROVED** [review]

**Comment by lemire** [comment]
Thanks @anonrig 

Merged.

Onward to more work.

**Pull request merged**